app/models/manageiq/providers/lenovo/physical_infra_manager/refresh_parser.rb

```diff
@@ -56,6 +57,14 @@ def get_physical_servers
       process_collection(nodes, :physical_servers) { |node| @parser.parse_physical_server(node) }
     end
 
+    def get_switches
+      return @all_switches if @all_switches
```

Comment:
Is this line needed?

Reply:
Maybe what you want here is to prevent a second call to xclarity API. As @felipedf said @all_switches is never initialized so this line is pointless, will make more sense if you do something like this:

```ruby
def get_switches
    @all_switches = @connection.discover_switches unless @all_switches

    process_collection(@all_switches, :switches) { |switch| @parser.parse_switch(switch) }
end
```

Reply:
Seems like what you want here is to prevent a second call to xclarity API, but as @felipedf said, the `@all_switches` is never initialized, so this line is pointless. Will make more sense if you do something like:

```ruby
def get_switches
    @all_switches = @connection.discover_switches unless @all_switches

    process_collection(@all_switches, :switches) { |switch| @parser.parse_switch(switch) }
end
```

Reply:
Seems like what you want here is to prevent a second call to xclarity client, but as @felipedf said the `@all_switches` is never initialized so this line is pointless. Will make more sense if you do something like this:
```ruby
def get_switches
     @all_switches = @connection.discover_switches unless @all_switches

    process_collection(@all_switches, :switches) { |switch| @parser.parse_switch(switch) }
end
```

Reply:
You are right. I am working on that.